File: src/main/java/aurocosh/divinefavor/common/area/WorldArea.java
package aurocosh.divinefavor.common.area;

import aurocosh.divinefavor.common.lib.math.CubeCoordinates;
import net.minecraft.util.math.BlockPos;

import java.util.HashSet;
import java.util.List;
import java.util.Set;

public class WorldArea {
    private Set<BlockPos> positions;
    private CubeCoordinates boundingBox;

    public WorldArea() {

        positions = new HashSet<>();
        boundingBox = new CubeCoordinates();
    }

    public void addPositions(List<BlockPos> posList) {
        positions.addAll(posList);
        refreshBoundingBox();
    }

    public void addPosition(BlockPos pos) {
        positions.add(pos);
        refreshBoundingBox();
    }

    public void removePosition(BlockPos pos) {
        positions.remove(pos);
        refreshBoundingBox();
    }

    public void clearPositions() {
        positions.clear();
        refreshBoundingBox();
    }

    private void refreshBoundingBox() {
        boundingBox = CubeCoordinates.getBoundingBox(positions);
    }

    public boolean isApartOfArea(BlockPos position) {
        return boundingBox.isCoordinateInside(position) && positions.contains(position);
    }
}

File: src/main/kotlin/aurocosh/divinefavor/common/area/WorldArea.kt
package aurocosh.divinefavor.common.area

import aurocosh.divinefavor.common.lib.math.CuboidBoundingBox
import net.minecraft.util.math.BlockPos
import java.util.*

class WorldArea {
    private val positions: MutableSet<BlockPos>
    private var boundingBox: CuboidBoundingBox

    init {
        positions = HashSet()
        boundingBox = CuboidBoundingBox()
    }

    fun addPositions(posList: List<BlockPos>) {
        positions.addAll(posList)
        refreshBoundingBox()
    }

    fun addPosition(pos: BlockPos) {
        positions.add(pos)
        refreshBoundingBox()
    }

    fun removePosition(pos: BlockPos) {
        positions.remove(pos)
        refreshBoundingBox()
    }

    fun clearPositions() {
        positions.clear()
        refreshBoundingBox()
    }

    private fun refreshBoundingBox() {
        boundingBox = CuboidBoundingBox.getBoundingBox(positions)
    }

    fun isApartOfArea(position: BlockPos): Boolean {
        return boundingBox.isCoordinateInside(position) && positions.contains(position)
    }
}
